packages/core/streams/openai-stream.test.tsx

```diff
@@ -365,6 +366,103 @@ describe('OpenAIStream', () => {
         '0:"."\n',
       ]);
     });
+
+    it('should call multiple tools when stream indicates multiple tool calls', async () => {
+      const data = new experimental_StreamData();
+
+      let appendedMessages: CreateMessage[] = [];
+      const stream = OpenAIStream(
+        await fetch(server.api + '/mock-tool-call', {
+          headers: {
+            'x-mock-service': 'openai',
+            'x-mock-type': 'tool_call',
+          },
+        }),
+        {
+          onFinal() {
+            data.close();
+          },
+          async experimental_onToolCall(
+            toolCallPayload,
+            appendToolCallMessages,
+          ) {
+            const { tools } = toolCallPayload;
+            for (const { id, func } of tools) {
+              data.append({ function_name: func.name });
+              if (func.name === 'get_date_time') {
+                const date = new Date(2023, 11, 17, 12, 31, 50, 69);
+                appendToolCallMessages({
+                  tool_call_id: id,
+                  function_name: func.name,
+                  tool_call_result: {
+                    date: date.toLocaleDateString(),
+                    time: date.toLocaleTimeString(),
+                  },
+                });
+              }
+              if (func.name === 'open_webpage') {
+                appendToolCallMessages({
+                  tool_call_id: id,
+                  function_name: func.name,
+                  tool_call_result: {
+                    url: 'https://openai.com',
+                  },
+                });
+              }
+            }
+            appendedMessages = appendToolCallMessages();
+            return 'experimental_onToolCall-return-value';
+          },
+          experimental_streamData: true,
+        },
+      );
+
+      const response = new StreamingTextResponse(stream, {}, data);
+
+      const client = createClient(response);
+      const chunks = await client.readAll();
+
+      expect(chunks).toEqual([
+        '2:[{"function_name":"get_date_time"},{"function_name":"open_webpage"}]\n',
+        '0:"experimental_onToolCall-return-value"\n',
```

Comment:
I'm a bit confused here. Wouldn't there be a `6:some-tool-call-information` message that's forwarded to the client when there is a tool call?

There is a similar test for functions:
```ts
    it('should send function response and data when onFunctionCall is defined, returns undefined, and data is added', async () => {
      const data = new experimental_StreamData();

      const stream = OpenAIStream(
        await fetch(server.api + '/mock-func-call', {
          headers: {
            'x-mock-service': 'openai',
            'x-mock-type': 'func_call',
          },
        }),
        {
          onFinal() {
            data.close();
          },
          async experimental_onFunctionCall({ name }) {
            data.append({ fn: name });

            // no response
          },
          experimental_streamData: true,
        },
      );

      const response = new StreamingTextResponse(stream, {}, data);

      const client = createClient(response);
      const chunks = await client.readAll();

      expect(chunks).toEqual([
        '2:[{"fn":"get_current_weather"}]\n',
        '1:{"function_call":{"name":"get_current_weather","arguments":"{\\n\\"location\\": \\"Charlottesville, Virginia\\",\\n\\"format\\": \\"celsius\\"\\n}"}}\n',
      ]);
    });
    ```
    
Ideally having a test that exactly mimics that for tools would be good, and then additional test for additional behavior.

Reply:
I thought so too.  But and I had it like that, but it was always returning 2 instead of 6.
It was too the point where I was thinking (Oh, maybe that number in these tests is the chunk number?)
I can certainly add a more basic tool call test that is like this one, I didn't see that one for function calling.

Reply:
Hm that's weird, it points to a potential issue. Do you see messages of type 6 delivered in the Chrome network monitor when you run your Svelte example?

Reply:
I figured this one out.  It's because I am returning a string from the function call, which causes it (by design) to send that string back to the client.  hence the `'0:"experimental_onToolCall-return-value"\n',`
and then the other 2: is from the `data` passed in .

I implemented another simpler test where the "server" doesn't handle the call, and so you get the new message back to the client with the 6: number.